xarray/backends/cfgrib_.py

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright 2017-2018 European Centre for Medium-Range Weather Forecasts (ECMWF).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+#   Alessandro Amici - B-Open - https://bopen.eu
+#
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from .. import Variable
+from ..core import indexing
+from ..core.utils import Frozen, FrozenOrderedDict
+from .common import AbstractDataStore, BackendArray
+from .file_manager import CachingFileManager
+from .locks import ensure_lock, SerializableLock
+
+# FIXME: Add a dedicated lock just in case, even if ecCodes is supposed to be thread-safe in most
+# circumstances. See: https://confluence.ecmwf.int/display/ECC/Frequently+Asked+Questions
+ECCODES_LOCK = SerializableLock()
+
+
+class CfGribArrayWrapper(BackendArray):
+    def __init__(self, backend_array):
+        self.backend_array = backend_array
+
+    def __getattr__(self, item):
+        return getattr(self.backend_array, item)
```

Comment:
This sort of forwarding is really error prone. Let's avoid if it in favor of a explicit solution if at all possible

Reply:
Done.